shockwave/trade/trade.go

```diff
@@ -19,10 +19,62 @@ func (offers Offers) Tradee() Offer {
 
 // Offer represents a user's offer in a trade.
 type Offer struct {
-	Name     string
-	Accepted bool
-	Items    []Item
+	Accepted  bool
+	UserId    int
+	ItemCount int
+	Items     []TradeItem
 }
 
-// Item is a type alias for [inventory.Item].
-type Item = inventory.Item
+type TradeItem struct {
+	ItemId     int
+	Type       string
+	Id         int
+	ClassId    int
+	Class      string
+	Colors     string
+	DimX, DimY int
+	Category   string
+	Groupable  int
+	Data       string
+	Day        int
+	Month      int
+	Year       int
+	SongId     int
+	SongName   string
+	SongDesc   string
+	PosterName string
+}
+
+func (item *TradeItem) Parse(p *g.Packet) {
```

Comment:
parsers should implement the Parsable interface https://github.com/xabbo/goearth/blob/8858d5e39165e20060978f81a8f729a61216ef50/packet.go#L25-L28
all reads must use the `Read*Ptr` variant passing `pos`, I plan on improving the interface to remove this requirement